NGitLab.Mock/Repository.cs

```diff
@@ -201,15 +201,35 @@ public Commit Commit(User user, string message, string targetBranch, IEnumerable
     public Commit Commit(CommitCreate commitCreate)
     {
         var repo = GetGitRepository();
-        var mustCreateBranch = !repo.Commits.Any();
-        if (!mustCreateBranch)
+        var branchExists = repo.Branches.Any(b => string.Equals(b.FriendlyName, commitCreate.Branch, StringComparison.Ordinal));
+
+        if (branchExists)
+        {
+            throw new GitLabBadRequestException($"A branch called '{commitCreate.Branch}' already exists.");
+        }
+
+        var isBranchEmpty = !repo.Commits.Any();
```

Comment:
It seems to me `isRepoEmpty` would be a more appropriate name here.

Reply:
updated